structures/segment-tree/problems/coming-and-going/coming-and-going.cpp

```diff
@@ -56,25 +58,31 @@ int build_segment_tree(int* const data, const unsigned int start, const unsigned
 	return tree[st_idx];
 }
 
-// #ifdef hackpackpp
-// update a value in the segment tree
-//
-// to support range sums, this function would have to be modified to recurse
-// down the tree to the leaf node before changing values; after the leaf node
-// is updated, the new value should be passed back as a return value and each
-// parent should update the sum they have by re-adding the values from the
-// two children it has
-//
-// tree        - the segment tree with values to be updated
-// start       - the start index of the _source_ array (should be 0)
-// end         - the end index of the _source_ array (size - 1)
-// changed_idx - the index of the element that changed
-// new_val     - the new value of the element at changed_idx
-// st_idx      - the current index of the segment tree (should be 0 when first called)
-// #endif
+
 void update_segment_tree(int* const tree, const unsigned int start, const unsigned int end, const unsigned int changed_idx, const int new_val, const unsigned int st_idx)
 {
 	// #ifdef hackpackpp
+	// update a value in the segment tree
+	//
+	// To support range sums, this function would have to be modified to recurse
```

Comment:
This isn't strictly relevant to the solving of the problem.  Instead of this long comment here, just include `segment-tree.cpp` in a section called "Reference" and put this comment there.